android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt

```diff
@@ -167,6 +174,18 @@ constructor(
     return if (output.hasActivity()) output else null
   }
 
+  // TODO refactor this code to remove hardcoded appended "PlanDefinition/"
```

Comment:
Is this TODO needed with the code implementation below in place? From the implementation below this todo, you are replacing the `instantiatesCanonical` when the first item does not contain a `/`.

Reply:
This is a TODO because the Instaciates Canonical is not limited to a PlanDef but also other resources , but for our usecase we mostly generate a CarePlan based on a defined PD see more here https://hl7.org/fhir/R4B/careplan-definitions.html#CarePlan.instantiatesCanonical

Reply:
Seems acceptable for now since this is only temporary. 

In line with this please link the ticket to track the actual issue/fix on this comment so that it is easy to know when the code can be stripped .  Something like :

```java
// TODO refactor this code to remove hardcoded appended PlanDefinition/  when this ticket is closed https://github.com/google/android-fhir/issues/<some ticket id>
```

Reply:
OK